src/cursor/cur_std.c

```diff
@@ -1254,10 +1254,21 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
             else
                 F_CLR(cursor, WT_CURSTD_BOUND_LOWER_INCLUSIVE);
             WT_ERR(__wt_buf_set(session, &cursor->lower_bound, key.data, key.size));
-        } else
-            WT_ERR_MSG(session, EINVAL,
-              "setting bounds only accepts \"upper\" or \"lower\" as the configuration");
-
+        } else if (WT_STRING_MATCH("prefix", cval.str, cval.len)) {
+            /* This configuration is special and sets both the lower and upper bound. */
+            WT_ERR(__wt_cursor_get_raw_key(cursor, &key));
+            WT_ERR(__wt_buf_set(session, &cursor->lower_bound, key.data, key.size));
+            WT_ERR(__wt_buf_set(session, &cursor->upper_bound, key.data, key.size));
+            /*
+             * Increment the given prefix for the upper bound. Make sure we only modify memory owned
+             * by us.
+             */
+            ((char *)cursor->upper_bound.data)[cursor->upper_bound.size - 1]++;
```

Comment:
This doesn't work for 'S' format tables as the string key can be length N and the string itself be "a\0". I'm investigating other possible solutions.

Reply:
In the case I saw in testing we had a size 10 key with the first two bytes being 'a\0' and I assume the remaining bytes were also '\0' I wonder if thats an optimization for short keys? The byte example is also concerning, I think what I need to do for 'S' is a strlen, and for 'u' is slightly more complex. Although I would've thought if you represented the byte string as a char* then added +1 it'll handle the wrapping for you. I'll write a test.

Reply:
Jie made a good point if your prefix was 0xFFFFFF and you added 1 then you'd end up with an invalid upper bound regardless of how you handled the addition.

Reply:
Nothing really comes after 0xFFFFFF without overflowing, so technically we can not define an exclusive upper bound. Since MongoDB is the end user for this feature and the keystrings have a defined ending, upper bounds are well defined in the execution layer. I am re-thinking my proposal of having the prefix bound in WiredTiger.

Reply:
If you have a prefix of 0xFFFFFF wouldn't you just unset the upper bound?  That would allow data from 0xFFFFFF00, 0xFFFFFF01 to 0xFFFFFFFF to 0xFFFFFFFFFFFFFFFFFFFFFFFF, etc.

Reply:
True, you could just unset the upper bound. My concern is that we're trying to handle a variety of edge cases, both around different Key format's and odd usage scenarios that I would prefer we don't as MongoDB can implement the logic on their side in a much safer way.